it/v2/endpoints/SubmitForeignPropertyBsasControllerISpec.scala

```diff
@@ -0,0 +1,617 @@
+        result.json shouldBe Json.toJson(RuleIncorrectPropertyAdjusted)
+      }
+
+      "validation error" when {
+        def validationErrorTest(requestNino: String, requestBsasId: String, requestBody: JsValue, expectedStatus: Int, expectedBody: MtdError): Unit = {
+          s"validation fails with ${expectedBody.code} error" in new Test {
+
+            override val nino: String = requestNino
+            override val bsasId: String = requestBsasId
+            override val requestBodyForeignPropertyJson: JsValue = requestBody
+
+            override def setupStubs(): StubMapping = {
+              AuditStub.audit()
+              AuthStub.authorised()
+              MtdIdLookupStub.ninoFound(requestNino)
+            }
+
+            val response: WSResponse = await(request().post(requestBodyForeignPropertyJson))
+            response.status shouldBe expectedStatus
+            response.json shouldBe Json.toJson(expectedBody)
+          }
+        }
+
+        val input = Seq(
+          ("Walrus", "f2fb30e5-4ab6-4a29-b3c1-c7264259ff1c", validRequestBody, BAD_REQUEST, NinoFormatError),
+          ("AA123456A", "Walrus", validRequestBody, BAD_REQUEST, BsasIdFormatError),
+          ("AA123456A", "f2fb30e5-4ab6-4a29-b3c1-c7264259ff1c", requestBodyAdjustmentValue, BAD_REQUEST, FormatAdjustmentValueError.withFieldName("/foreignFhlEea/expenses/consolidatedExpenses")),
+          ("AA123456A", "f2fb30e5-4ab6-4a29-b3c1-c7264259ff1c", requestBodyRangeInvalid, BAD_REQUEST, RuleAdjustmentRangeInvalid.withFieldName("/foreignFhlEea/expenses/consolidatedExpenses")),
+          ("AA123456A", "f2fb30e5-4ab6-4a29-b3c1-c7264259ff1c", requestBodyIncorrectBody, BAD_REQUEST, RuleIncorrectOrEmptyBodyError),
+          ("AA123456A", "f2fb30e5-4ab6-4a29-b3c1-c7264259ff1c", requestBodyBothExpenses, BAD_REQUEST, RuleBothExpensesError)
+        )
+
+        input.foreach(args => (validationErrorTest _).tupled(args))
+      }
+
+      "des service error" when {
+        def serviceErrorTest(desStatus: Int, desCode: String, expectedStatus: Int, expectedBody: MtdError): Unit = {
+          s"des returns an $desCode error and status $desStatus" in new Test {
+
+            override def setupStubs(): StubMapping = {
+              AuditStub.audit()
+              AuthStub.authorised()
+              MtdIdLookupStub.ninoFound(nino)
+              DesStub.onError(DesStub.PUT, desUrl, desStatus, errorBody(desCode))
+            }
+
+            val response: WSResponse = await(request().post(requestBodyForeignPropertyJson))
+            response.status shouldBe expectedStatus
+            response.json shouldBe Json.toJson(expectedBody)
+          }
+        }
+
+        val input = Seq(
+          (BAD_REQUEST, "INVALID_TAXABLE_ENTITY_ID", BAD_REQUEST, NinoFormatError),
+          (BAD_REQUEST, "INVALID_CALCULATION_ID", BAD_REQUEST, BsasIdFormatError),
+          (BAD_REQUEST, "INVALID_PAYLOAD", INTERNAL_SERVER_ERROR, DownstreamError),
```

Comment:
can you check this with francis the spec has `INVALID_INVALID_PAYLOAD` not `INVALID_PAYLOAD`
this will need updating in the service as well if this is wrong

Reply:
Confirmed as the spec being wrong